crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -432,20 +432,37 @@ impl EventHandler {
                     "Intercepted HTLC"
                 );
 
-                // if we have already a channel with them, we try to forward the payment.
-                // TODO: here we would need to increase the channel size if the channel is too small
-                if let Some(channel) =
-                    self.channel_manager
-                        .list_channels()
-                        .iter()
-                        .find(|channel_details| {
-                            if let Some(scid) = channel_details.short_channel_id {
-                                scid == requested_next_hop_scid
-                            } else {
-                                false
+                let target_node_id = {
+                    let fake_channel_payments = self.fake_channel_payments.lock().unwrap();
+                    match fake_channel_payments.get(&requested_next_hop_scid) {
```

Comment:
Do I understand it correctly that `intercept_htlc` is really just being called for `fake_channel_ids` and for `known_channel_ids` it is not being called?

Reply:
> But even if we did support "regular" invoices, the matching HTLC should not trigger this event and should be handled differently (I think automatically by LDK).

Cool, I did not know this. 

> The app currently only supports interceptable invoices

I thought I had added logic to decided wether to go for intercept able invoices or _normal_ invoices.

Reply:
It looks like the app only supports interceptable invoices currently:

https://github.com/get10101/10101/blob/e2552992fb3ad2bdbd9bd954ef9c002b6099bdf0/mobile/native/src/ln_dlc/mod.rs#L313-L346

We could change that in the future, although I think @holzeis had reservations.

Reply:
I think we need both implementations eventually. I don't think we can go without that fix as it is not always possible for the app to know what invoice to create. e.g. assume the user creates an interceptable invoice (since no channel exists yet). Before that invoice has been paid, the user creates a second invoice. At this point the app has not enough information to create either a regular invoice (assuming the interceptable invoice will be paid first) or another interceptable invoice (assuming the first invoice does not get paid).

Eventually we should go with a fix that does not create a interceptable invoice if a channel already exists, but this is not critical for now. That's why this fix on focused on ensuring that multiple interceptable invoices will not lead to multiple channels.